pontoon/base/fluent.py

```diff
@@ -158,3 +159,53 @@ def get_simple_preview(content):
         tree = translation_ast.attributes[0]
 
     return serialize_value(tree.value)
+
+
+def is_plural_expression(expression):
+    from pontoon.base.models import Locale
+
+    CLDR_PLURALS = [c for _, c in Locale.CLDR_PLURALS]
+
+    if isinstance(expression, ast.SelectExpression):
+        return all(
+            isinstance(variant.key, ast.NumberLiteral)
+            or variant.key.name in CLDR_PLURALS
+            for variant in expression.variants
+        )
+
+    return False
+
+
+def create_locale_plural_variants(node, locale):
+    if not is_plural_expression(node):
+        return
+
+    numeric = []
+    source_plurals = {}
+    locale_plurals = []
+    default = None
+
+    for variant in node.variants:
+        key = variant.key
+        if isinstance(key, ast.NumberLiteral):
+            numeric.append(variant)
+        elif isinstance(key, ast.Identifier) and key.name in locale.cldr_plurals_list():
+            source_plurals[key.name] = variant
+        if variant.default:
+            default = variant
+
+    for plural in locale.cldr_plurals_list().split(", "):
+        if plural in source_plurals.keys():
+            variant = source_plurals[plural]
+        else:
+            variant = copy.deepcopy(default)
+            variant.default = False
+            variant.key.name = plural
+        locale_plurals.append(variant)
+
+    variants = numeric + locale_plurals
+
+    if len([v for v in variants if v.default]) == 0:
+        variants[-1].default = True
```

Comment:
This results in the wrong default case for some Slavic languages. For plurals, the default should be defined by the locale rather than the source message.

If this isn't possible right now, then a code comment somewhere around this function should note this need for later improvement.

Reply:
How do we know which is the default variant for the locale?

Reply:
It should be the last entry in `cldr_plurals_list()`, as long as that's either `many` or `other`. If it's neither, then we've bad data and really ought to either fix the data or add an `other` at the end.

Reply:
That sounds like we should always just use the last plural category (implemented in [1f64847](https://github.com/mozilla/pontoon/pull/2802/commits/1f648473290c5fc143e148f2b6d88db2ec5a4395)) or fix data (filed https://github.com/mozilla/pontoon/issues/2807).